.github/workflows/testsuite.yml

```diff
@@ -6,14 +6,26 @@ on:
 jobs:
   tests:
     runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macOS-latest, windows-latest]
         python-version: [3.7, 3.8, 3.9, "3.10", "3.11"]
+        experimental: [false]
         include:
-          - python-version: "pypy-3.7"
-            os: ubuntu-latest
+          - os: ubuntu-latest
+            python-version: "pypy-3.7"
+            experimental: false
+          - os: ubuntu-latest
+            python-version: "3.12-dev"
+            experimental: true
+          - os: macOS-latest
+            python-version: "3.12-dev"
+            experimental: true
+          - os: windows-latest
+            python-version: "3.12-dev"
+            experimental: true
```

Comment:
I would probably just disable the tests with extra requirements for 3.12 (by adding a version check to the respective steps). It is common that larger packages do not support a new Python in early development stages. We can remove the check later.